solr/core/src/java/org/apache/solr/handler/IndexFetcher.java

```diff
@@ -332,24 +336,26 @@ private void setLeaderUrl(String leaderUrl) {
         solrCore
             .getCoreContainer()
             .getAllowListUrlChecker()
-            .checkAllowList(Collections.singletonList(leaderUrl), clusterState);
+            .checkAllowList(Collections.singletonList(leaderCoreUrl), clusterState);
       } catch (MalformedURLException e) {
         throw new SolrException(
-            SolrException.ErrorCode.SERVER_ERROR, "Malformed 'leaderUrl' " + leaderUrl, e);
+            SolrException.ErrorCode.SERVER_ERROR, "Malformed 'leaderUrl' " + leaderCoreUrl, e);
```

Comment:
should this variable be called leaderUrl, or the message changned to leaderCoreUrl?